addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1370,6 +1378,87 @@ MockServer.include({
         messages.length = Math.min(messages.length, limit);
         return this._mockMailMessageMessageFormat(messages.map(message => message.id));
     },
+    /**
+     * Simulates the `/mail/message/add_reaction` route.
+     *
+     * @private
+     * @param {integer} message_id
+     * @param {string} content
+     * @returns {Object}
+     */
+    _mockRouteMailMessageAddReaction(message_id, content) {
+        const partner_id = this.currentPartnerId;
+        this._mockMailMessage_MessageAddReaction(message_id, content);
+        const message_reactions = this._getRecords('mail.message.reaction', [
+            ['message_id', '=', message_id],
+            ['content', '=', content],
+        ]);
+        return {
+            id: message_id,
+            messageReactionGroups: [['insert', {
+                'messageId': message_id,
+                'content': content,
+                'count': message_reactions.length,
+                'partners': [['insert-and-replace', [{ id: partner_id }]]],
+            }]],
```

Comment:
Also lacking handling of guests. Should be done so that making tests with guests and reaction just work as expected.

Reply:
According to @seb-odoo, it seems we don't have a correct test env for guests at the moment

Reply:
Would like small explanation because I don't get the problem @seb-odoo

Reply:
Note: and then it would be nice to add one extra test, actually testing the guest case too.